### PR TITLE
Select field custom picker

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**

### CODE REVIEW

File: src/demos/select-field/select-field-demo-custom-picker.component.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<sky-modal>
+  <sky-modal-header>
+    {{ title }}
+  </sky-modal-header>
+  <sky-modal-content>
+    <ng-container
+      *ngIf="context.pickerType === 'color'; else fruitPicker"
```
Comment (Blackbaud-AlexKingman) [inline src/demos/select-field/select-field-demo-custom-picker.component.html]: Code style > Structural directives should be on the same line (I see a few of those in this template).

Comment (Blackbaud-AlexKingman) [inline src/demos/select-field/select-field-demo-custom-picker.component.html]: Code style > Structural directives should be on the same line (I see a few of those in this template).

File: src/demos/select-field/select-field-demo-custom-picker.component.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      >
+        {{ fruitMap[fruit.id] }}
+      </button>
+    </ng-template>
+  </sky-modal-content>
+  <sky-modal-footer *ngIf="context.pickerType === 'fruit'">
+    <button
+      class="sky-btn sky-btn-primary"
+      type="button"
+      (click)="saveFruits()">
```
Comment (Blackbaud-AlexKingman) [inline src/demos/select-field/select-field-demo-custom-picker.component.html]: Code style > Closing brackets should be on their own line.

File: src/demos/select-field/select-field-demo-custom-picker.component.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    '2': '🍌',
+    '3': '🍈',
+    '4': '🍇',
+    '5': '🍋',
+    '6': '🍓'
+  };
+
+  constructor(
+    private instance: SkyModalInstance,
+    public context: SkySelectFieldDemoCustomPickerContext
```
Comment (Blackbaud-AlexKingman) [inline src/demos/select-field/select-field-demo-custom-picker.component.ts]: Any reason this needs to be a `public` property?

File: src/demos/select-field/select-field-demo-custom-picker.component.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  ) { }
+
+  public ngOnInit() {
```
Comment (Blackbaud-AlexKingman) [inline src/demos/select-field/select-field-demo-custom-picker.component.ts]: Code style > functions need a return type (There's a few in here).

File: src/demos/select-field/select-field-demo.component.html [hunk 5]
```diff
@@ -142,3 +142,44 @@ <h2 class="sky-section-heading">
...
+<sky-row>
+  <sky-column screenSmall="12">
+    <h2 class="sky-section-heading">
+      Custom picker
+    </h2>
+
+    <div class="sky-form-group">
+      <label
+        class="sky-control-label"
+        id="my-select-field-custom-1">
```
Comment (Blackbaud-AlexKingman) [inline src/demos/select-field/select-field-demo.component.html]: Code style > closing brackets need to be on their own line (there's a few in here).

File: src/demos/select-field/select-field-demo.component.html [hunk 6]
```diff
@@ -142,3 +142,44 @@ <h2 class="sky-section-heading">
...
+      </label>
+      <sky-select-field
+        [data]="colorStream"
+        ariaLabelledBy="my-select-field-custom-1"
+        selectMode="single"
+        singleSelectClearButtonTitle="Clear selected color"
+        singleSelectOpenButtonTitle="Select color"
+        singleSelectPlaceholderText="Select color..."
+        name="favoriteColorCustom"
+        [customPicker]="customFavoriteColorPicker">
```
Comment (Blackbaud-AlexKingman) [inline src/demos/select-field/select-field-demo.component.html]: Code style > check your ordering of inputs here.
https://host.nxt.blackbaud.com/skyux-team-docs/style#component-html-tags

File: src/demos/select-field/select-field-demo.component.ts [hunk 7]
```diff
@@ -79,4 +113,29 @@ export class SkySelectFieldDemoComponent implements OnInit {
...
+          useValue: {
+            pickerContext,
+            pickerType
+          }
+        }
+      ]
+    );
+
+    picker.closed.subscribe((args: SkyModalCloseArgs) => {
+      if (args.reason === 'save') {
```
Comment (Blackbaud-AlexKingman) [inline src/demos/select-field/select-field-demo.component.ts:136]: Should we be covering the "closed" route as well? Here's what I was running into:
 
1. Open the multiselect picker
2. Make some selections
3. Click 'close' (NOT select).
4. Notice tokens are not updated, which is expected.
5. Open the multiselect picker again
6. Notice your old selections are still selected.

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Looks pretty good - just a few code style items and one other question. I'm happy to take care of the code style stuff if need be. Thanks!

File: src/demos/select-field/select-field-demo-custom-picker.component.html (hunk 1 above)
Comment (Blackbaud-PaulCrowder) [reply src/demos/select-field/select-field-demo-custom-picker.component.html]: Fixed.

**Review by Blackbaud-AlexKingman — DISMISSED** [review]
LGTM. Also, thanks for cleaning up the code style on the old demo.

**Pull request merged**